File: src/codility/lesson5/PassingCars.java
package codility.lesson5;

public class PassingCars {

    public int solution(int[] A) {
        int westCars = 0;
        int ans = 0;

        for (int i = A.length-1;i>=0;i--) {
            if (A[i] == 0 ) {
                ans += westCars;

                if (ans > 1_000_000_000) {
                    return -1;
                }
            }
            else {
                westCars++;
            }
        }
        return ans;

    }

    public int solution1(int[] A) {
        int westCar =0;

        for (int a : A) {
            if (a == 1) {
                westCar++;
            }
        }

        int result = 0;
        for (int a : A) {
            if (a == 0) {
                result += westCar;

                if (result > 1000000000) {
                    return -1;
                }
            }
            else {
                westCar--;
            }
        }

        return result;
    }
}

File: src/codility/kotlin/PassingCars.kt
package codility.kotlin

fun main() {
    PassingCars().solution(intArrayOf(0,1,0,1,1))
}

class PassingCars {



    fun solution(A: IntArray): Int {
        // write your code in Kotlin
        var result = 0
        var westCarCount = 0

        for (i in A.size - 1 downTo 0) {
            if (A[i] == 1) {
                westCarCount++
            } else {
                result += westCarCount
                if (result > 1_000_000_000) {
                    return -1
                }
            }
        }
        return result
    }

}